### PR TITLE
Timeline Backend Functionality

### PR DESCRIPTION
**Pull request by nolantuite**
Description: Timeline functionality 

Trello Card Link: https://trello.com/c/rBwXEMyx/45-develop-timeline-collection-for-users-feed

### CODE REVIEW

File: backend/src/routes/post.ts [hunk 1]
```diff
@@ -22,18 +24,25 @@ router.get('/:postID', async (ctx, next) => {
 	}
 	ctx.body = post;
 });
+
 router.post('/', async (ctx, next) => {
 	const post = new Post();
 	post.id = new mongodb.ObjectId();
 	post.author = null; //This is dependent on Auth to determine the author
```
Comment (baylee-rl) [inline backend/src/routes/post.ts]: make sure to update auth stuff or make a trello card for it

File: backend/src/routes/user.ts [hunk 2]
```diff
@@ -17,26 +18,109 @@ router.get('/:username', async (ctx, next) => {
...
+	if (user === null) {
+		ctx.status = 404;
+		return;
+	}
+	ctx.body = user.followers;
+});
+
+router.get('/timeline', async (ctx, next) => {
+	const user = await User.findOneBy({
+		username: ctx.state.user.id,
```
Comment (baylee-rl) [inline backend/src/routes/user.ts]: `id` won't be the username, it'll be the object id

File: backend/src/routes/post.ts [hunk 3]
```diff
@@ -59,9 +68,20 @@ router.post('/:postID/unlike', async (ctx, next) => {
 	ctx.body = post;
 });
 
+//delete post by postID
+router.delete('/:postID', async (ctx, next) => {
+	const post = Post.findOneBy(mongodb.ObjectId(ctx.params.postID));
+	//get author of post
+	const author = User.findOneBy(mongodb.ObjectId((await post).author));
```
Comment (shreyasminocha) [inline backend/src/routes/post.ts]: Could you prefix the right-hand sides of these assignments with `await`? Then we wouldn't need to prefix `post` and `author` with `await` every time we use them.

**Review by shreyasminocha — CHANGES_REQUESTED** [review]

**Review by baylee-rl — CHANGES_REQUESTED** [review]
just a couple minor things!

**Pull request merged**